akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala

```diff
@@ -785,6 +795,13 @@ abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(confi
       expectMsg("x-b-2")
       expectNoMsg(100.millis)
     }
+    "support calling persistAll with Nil" in {
+      val persistentActor = namedPersistentActor[PersistAllNilPersistentActor]
+      persistentActor ! Cmd("x")
+      expectMsg("before-nil")
+      expectMsg("after-nil")
+      expectMsg("x")
```

Comment:
Well, now I do remember being part of the decision to do this optimization—why go through the journal when that is not needed. I’m not sure it’s bad enough to ticketify, let’s watch out for related user feedback.